lib/connections/service.go

```diff
@@ -370,15 +382,16 @@ func (s *Service) connect() {
 	}
 }
 
-func (s *Service) shouldLimit(addr net.Addr) bool {
-	if s.cfg.Options().LimitBandwidthInLan {
+func (s *Service) shouldLimit(addr net.Addr, options config.OptionsConfiguration) bool {
+	if options.LimitBandwidthInLan {
```

Comment:
If there was more than one thing, which relied on the config to make a decision in shouldLimit, I'd probably be fine with it, but given there is not more than one thing, this just seems like a premature optimization (or rather design decision) without a good reason... Just in case we'd ever had more than one thing relying on the config to make that decision.